packages/kbn-es/src/cluster.js

```diff
@@ -254,8 +254,8 @@ exports.Cluster = class Cluster {
       'ingest.geoip.downloader.enabled=false',
     ].concat(options.esArgs || []);
 
-    // Add to esArgs if ssl is enabled
-    if (this._ssl) {
+    // Add to esArgs if ssl is enabled only if `http.ssl` config isn't provided already.
+    if (this._ssl && !esArgs.some((arg) => arg.startsWith('xpack.security.http.ssl.'))) {
```

Comment:
__note:__ it seemed like the simplest possible change to support what we need, but happy to accept any other suggestions.